src/encoded/types/shared_calculated_properties.py

```diff
@@ -482,85 +482,85 @@ class CalculatedReplicationType:
             if tech_rep_num < bio_rep_dict[bio_rep_num]['technical_replicate_number']:
-                bio_rep_dict[bio_rep_num] = replicate_object
-
-        # Compare the biosamples to see if for humans they are the same donor and for
-        # model organisms if they are sex-matched and age-matched
-        biosample_donor_list = []
-        biosample_number_list = []
-
-        for replicate_object in bio_rep_dict.values():
-            if 'libraries' in replicate_object and replicate_object['libraries']:
-                biosamples = request.select_distinct_values(
-                    'biosample', *replicate_object['libraries']
-                )
-                if biosamples:
-                    for b in biosamples:
-                        biosample_object = request.embed(b, '@@object')
-                        biosample_donor_list.append(
-                            biosample_object.get('donor')
-                        )
-                        biosample_number_list.append(
-                            replicate_object.get('biological_replicate_number')
-                        )
-                        biosample_species = biosample_object.get('organism')
-                        biosample_type_object = request.embed(
-                            biosample_object['biosample_ontology'],
-                            '@@object'
-                        )
-                        biosample_type = biosample_type_object.get('classification')
-                else:
-                    # special treatment for "RNA Bind-n-Seq" they will be called unreplicated
-                    # untill we change our mind
-                    if assay_term_name == 'RNA Bind-n-Seq':
-                        return 'unreplicated'
-                    # If I have a library without a biosample,
-                    # I cannot make a call about replicate structure
-                    return None
-            else:
-                # REPLICATES WITH NO LIBRARIES WILL BE CAUGHT BY AUDIT (TICKET 3268)
-                # If I have a replicate without a library,
-                # I cannot make a call about the replicate structure
+                bio_rep_dict[bio_rep_num] = r
+        biosample_donors_set = set()
+        biosample_numbers_set = set()
+        for r in bio_rep_dict.values():
+            library = r.get('library')
+            if not library:
+                # If I have a replicate without a library I cannot make a
+                # call about the replicate structure.
                 return None
-
-        #  exclude ENCODE2
-        if (len(set(biosample_number_list)) < 2):
+            library_object = request.embed(library, '@@object')
+            biosample = library_object.get('biosample')
+            if not biosample:
+                # Special treatment for "RNA Bind-n-Seq" they will be called
+                # unreplicated until we change our mind.
+                if assay_term_name == 'RNA Bind-n-Seq':
```

Comment:
I have a hard time understanding this. You're trying to say any RNA Bind-n-Seq experiment will be reported as unreplicated? If that's the case, this should go right at the top of the function, so the calculation is cheaper for such experiments. The bonus is also that it is clearer when reading the function that there is a special case.

I also think that this special behavior should be documented in the function docstring.

Reply:
RNA BindNseq can be declared as unreplicated if it has libraries in replicates with files, otherwise it is "unknown" replication type

Reply:
Thanks for adding to docstring